edx_rest_framework_extensions/auth/jwt/decoder.py

```diff
@@ -168,13 +168,12 @@ def _decode_and_verify_token(token, jwt_issuer):
         'verify_exp': api_settings.JWT_VERIFY_EXPIRATION,
         'verify_aud': settings.JWT_AUTH.get('JWT_VERIFY_AUDIENCE', True),
         'verify_iss': False,  # TODO (ARCH-204): manually verify until issuer is configured correctly.
-        'verify_signature': False,  # Verified with JWS already
+        'verify_signature': api_settings.JWT_VERIFY
     }
 
     decoded_token = jwt.decode(
         token,
         jwt_issuer['SECRET_KEY'],
-        api_settings.JWT_VERIFY,
```

Comment:
In 1.1.7 this variable was settings `verify=True` [here](https://github.com/jpadilla/pyjwt/blob/1.7.1/jwt/api_jwt.py#L71)  and in `decode` method it was doing  multiple things and most important was following.
```
if verify:
    merged_options = merge_dict(self.options, options)
    self._validate_claims(payload, merged_options, **kwargs)
```
I replaced `'verify_signature': api_settings.JWT_VERIFY` due to this [change](https://github.com/jpadilla/pyjwt/pull/271/files#diff-6893ad4a1c5a36b8af3028db8c8bc3b62418149843fc382faf901eaab008e380R63) but I think it should be more explicit like `'verify_signature': True`

With True it will check `self._validate_claims` as per [2.0.1](https://github.com/jpadilla/pyjwt/blob/2.0.1/jwt/api_jwt.py#L98) 
 
Please let me know If it makes any sense or more debugging require ?

Reply:
I'm not sure.  The earlier code read:
```
'verify_signature': False, # Verified with JWS already
```
The reason we weren't verifying signature here is because JWS is verifying elsewhere. I'm guessing that this was for potential performance reasons, to not verify twice, but not sure.

Some questions:
1. Does not verifying again actually provide enough performance impact to warrant trying to do it once, or does this just make the code riskier than necessary? I think it takes more time to create a signature, but not much to verify it.
2. Is JWS actually performing the same verification, or is it slightly different? This only matters if we think there is a performance issue of doing the work twice.

I'd lean toward verifying (the way you have it), but just want to find some way to ensure we aren't introducing a performance issue. We could also ask @nasthagiri.

Reply:
@nasthagiri Please review this PR.

Reply:
@awais786: I think this looks correct, meaning I don’t think this will change behavior. However:
1. I don’t see any legacy test for signature verification. Did I just miss it, or can you point it out? Maybe it isn’t in the test name, but could be added to a comment?
2. Do all JWTs use JWS? I’d like to ensure, hopefully through a test, that a JWT could never pass with an invalid signature.

Reply:
@robrap These [tests](https://github.com/edx/edx-drf-extensions/blob/master/edx_rest_framework_extensions/auth/jwt/tests/test_decoder.py#L69-L100) are testing signatures. I have also added 1 more test. Does this answer both of your questions ?